File: java/ArqLibraryClassic/ArqLibraryLayered/src/bnsit/ms/arq/library/Generated.java
package bnsit.ms.arq.library;

import java.util.HashMap;
import java.util.Map;

public class Generated {

    private static final Map<String, Long> ids = new HashMap<>();

    public static long bookId()
    {
        return idFor("book");
    }

    public static long borrowingId()
    {
        return idFor("borrowing");
    }

    public static long userId()
    {
        return idFor("user");
    }

    public static void resetUserId()
    {
        resetIdFor("user");
    }

    public static void resetBorrowingId()
    {
        resetIdFor("borrowing");
    }

    public static void resetBookId()
    {
        resetIdFor("book");
    }

    private static long idFor(String name)
    {
        if (ids.containsKey(name))
        {
            long id = ids.get(name);
            ids.put(name, increment(id));
        }
        else
        {
            long id = 1;
            ids.put(name, id);
        }

        return ids.get(name);
    }

    private static long increment(long id)
    {
        return id + 1;
    }

    private static void resetIdFor(String name)
    {
        ids.put(name, 0L);
    }
}

File: dotnet/ArqLibrarianClassic/ArqLibrarianClassic/BorrowingContext/Generated.cs
﻿using System.Collections.Generic;

namespace ArqLibrarianClassic.BorrowingContext
{
    public static class Generated
    {
        private static readonly Dictionary<string, long> ids = new Dictionary<string, long>();
        
        public static long BookId()
        {
            return IdFor("book");
        }

        public static long BorrowingId()
        {
            return IdFor("borrowing");
        }
        
        public static long UserId()
        {
            return IdFor("user");
        }

        public static void ResetUserId()
        {
            ResetIdFor("user");
        }

        public static void ResetBorrowingId()
        {
            ResetIdFor("borrowing");
        }

        public static void ResetBookId()
        {
            ResetIdFor("book");
        }

        private static long IdFor(string name)
        {
            if (ids.ContainsKey(name))
            {
                var id = ids[name];
                ids[name] = Increment(id);
            }
            else
            {
                long id = 1;
                ids[name] = id;
            }

            return ids[name];
        }

        private static long Increment(long id)
        {
            return id + 1;
        }

        private static void ResetIdFor(string name)
        {
            ids[name] = 0;
        }
    }
}